summingbird-scalding/src/main/scala/com/twitter/summingbird/scalding/batch/BatchedStore.scala

```diff
@@ -96,10 +104,9 @@ trait BatchedStore[K, V] extends scalding.Store[K, V] { self =>
         // make sure we checkpoint to disk to avoid double computation:
         val checked = if (batches.size > 1) lastVals.forceToDisk else lastVals
         batches.foreach { batchID =>
-          val thisBatch = checked.filter {
-            case (b, kv) =>
-              (b == batchID) && !pruning.prune(kv, batcher.latestTimeOf(b))
-          }
+          // closures are the devil
+          val filterFn = WriteFilter(batchID, batcher.latestTimeOf(batchID), pruning)
```

Comment:
Curiosity: why create a new class instead of isolating captured values using local `val`s?

Reply:
just to be 100% clear about what is captured. a side benefit is that is has a good equality, but I doubt that will come into play.

I was surprised to see what was captured here because you have 3 nested layers of closure.

Reply:
👍 thanks for the clarification. I think we could do better regarding capturing. We could do something at runtime like Spark does or even try to inspect the function using a macro to detect captured values that aren't serializable.

Reply:
given that lightbend are trying to remove macros, or at least drastically narrow the scope, I'm a little worried about being too macro reliant. Note that scalacenter's spores are not even available on 2.12 due to issues with macro interaction with a new version.